src/append/rolling_file/policy/compound/trigger/time.rs

```diff
@@ -236,14 +257,19 @@ impl TimeTrigger {
             let weekday = current.weekday().num_days_from_monday() as i64; // Monday is the first day of the week
             let time = Local.with_ymd_and_hms(year, month, day, 0, 0, 0).unwrap();
             let increment = if modulate { n - week0 % n } else { n };
-            return time + Duration::weeks(increment) - Duration::days(weekday);
+            let dur = Duration::try_weeks(increment)
+                .ok_or(TimeTrigerIntervalError::Week(interval))?
+                - Duration::try_days(weekday).ok_or(TimeTrigerIntervalError::Day(interval))?;
```

Comment:
The `ok_or` in `try_days(weekday).ok_or()` is unnecessary, because the `weekday` will always be valid. `unwrap()` is enough. But it's also good to me if you want to use `ok_or`.